codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/WebSocketHAProxyHandshaker.java

```diff
@@ -0,0 +1,231 @@
+import static io.netty.handler.codec.http.HttpHeaderNames.WEBSOCKET_ORIGIN;
+import static io.netty.handler.codec.http.HttpHeaderNames.WEBSOCKET_PROTOCOL;
+import static io.netty.handler.codec.http.HttpHeaderNames.WEBSOCKET_LOCATION;
+import static io.netty.handler.codec.http.HttpHeaderNames.ORIGIN;
+import static io.netty.handler.codec.http.HttpHeaderValues.WEBSOCKET;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMessage;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpObjectDecoder;
+import io.netty.handler.codec.http.HttpRequestDecoder;
+import io.netty.handler.codec.http.HttpResponseEncoder;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.codec.http.websocketx.WebSocketHandshakeException;
+import io.netty.handler.codec.http.websocketx.WebSocketServerHandshaker00;
+import io.netty.handler.codec.sockjs.handler.SessionHandler.Event;
+import io.netty.util.AsciiString;
+import io.netty.util.concurrent.FastThreadLocal;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.regex.Pattern;
+
+/**
+ * An extension of {@link WebSocketServerHandshaker00} which handles Hixie76
+ * upgrade requests that plays nicely with HAProxy.
+ *
+ * When HAProxy sends an WebSocket Hixie 76 upgrade reqeust will first send the request headers,
+ * after receiveing the response, it will then send the nouce.
+ *
+ * This class will extract the 'SEC_WEBSOCKET_KEY1' and 'SEC_WEBSOCKET_KEY2' from the first request
+ * and later use it for the actual handshake.
+ *
+ * Note that currently this does not work as desired with unless the HTTP request header 'Content-Lenght'
+ * has been set. Netty's {@link HttpObjectDecoder} uses
+ * {@link HttpHeaders#getContentLength(HttpMessage, long)} which will set the lenght
+ * of Hixie 76 request to '8'. But there will be no body in the first request and therefor the message will be
+ * dropped.
+ */
+public class WebSocketHAProxyHandshaker extends WebSocketServerHandshaker00 {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(WebSocketHAProxyHandshaker.class);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) "logger" is the name of a field in "WebSocketServerHandshaker". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2387)